src/libraries/System.Private.CoreLib/src/System/Buffers/SharedArrayPool.cs

```diff
@@ -493,6 +493,9 @@ public void Trim(int currentMilliseconds, int id, Utilities.MemoryPressure press
 
     internal static class SharedArrayPoolStatics
     {
+        /// <summary>Whether the diagnostic pool is enabled.</summary>
+        internal static readonly bool s_diagnosticPoolEnabled =
```

Comment:
I think that we would want this trimmed by default for PublishTrimmed / PublishAot.